packages/xdl/src/Web.js

```diff
@@ -79,39 +121,126 @@ export async function onlySupportsWebAsync(projectRoot) {
   return false;
 }
 
-export async function ensureWebSupportAsync(projectRoot) {
+async function promptToAddWebPlatform() {
+  return await promptAsync(
+    `It appears you don't explicitly define "${chalk.underline(
+      `web`
+    )}" as one of the supported "${chalk.underline(`platforms`)}" in your project's ${chalk.green(
+      `app.json`
+    )}. \n  Would you like to add it?`
+  );
+}
+
+async function installAsync(projectRoot, libraries = []) {
+  const options = {
+    cwd: projectRoot,
+    stdio: 'inherit',
+  };
+  try {
+    if (await ConfigUtils.isUsingYarn(projectRoot)) {
+      await spawnAsync('yarn', ['add', ...libraries], options);
+    } else {
+      await spawnAsync('npm', ['install', '--save', ...libraries], options);
+    }
+  } catch (error) {
+    throw new XDLError(
+      XDLError.WEB_NOT_CONFIGURED,
+      'Failed to install libraries: ' + error.message
+    );
+  }
+}
+
+function areModulesInstalled(projectRoot, exp) {
+  const modulesPath = ConfigUtils.resolveModule(`.`, projectRoot, exp);
+  return fs.existsSync(modulesPath);
+}
+
+async function getLibraryVersionAsync(projectRoot, exp, packageName) {
+  const possibleModulePath = ConfigUtils.resolveModule(
+    `${packageName}/package.json`,
+    projectRoot,
+    exp
+  );
+  if (fs.existsSync(possibleModulePath)) {
+    const packageJson = await JsonFile.readAsync(possibleModulePath);
+    return packageJson.version;
+  }
+}
+
+export async function ensureWebSupportAsync(projectRoot, isInteractive = true) {
+  let errors = [];
+
   const hasWebSupport = await hasWebSupportAsync(projectRoot);
   if (!hasWebSupport) {
-    throw new XDLError(ErrorCode.WEB_NOT_CONFIGURED, getWebSetupLogs());
+    if (isInteractive && (await promptToAddWebPlatform())) {
+      await ConfigUtils.addPlatform(projectRoot, 'web');
+    } else {
+      errors.push(getWebSetupLogs());
+      isInteractive = false;
```

Comment:
bail out here instead of continuing on?

Reply:
In the case where you have more than one error, it'd be nice to get them all at once rather then fixing an error and running only to get a new error.